contracts/limitOrder/LimitOrderBook.sol

```diff
@@ -30,6 +33,8 @@ contract LimitOrderBook is
     using PerpMath for uint256;
     using PerpSafeCast for uint256;
     using SignedSafeMathUpgradeable for int256;
+    using SafeMathUpgradeable for uint256;
+    using SafeMathUpgradeable for uint8;
 
     // NOTE: cannot use `OrderType orderType` here, use `uint8 orderType` instead
     // solhint-disable-next-line func-name-mixedcase
```

Comment:
I'm not really familiar with pre-computing hash but I thought we could compute this locally and paste the hash directly and can thus save this long line

Reply:
Yes, we could. Though I think this is less error-prone since we might forget to change the hash somehow. Also more readable than `54ea8c184890b5a7f7321f45a2ae952f0af50e3467b2216418a683566bf57c30`.

Reply:
Hmm but I think the point of pre-computing hash is to save the gas of calculating that line `keccak256(..)`.
But yeah you're right that this might be prone to error...

I guess we can keep a line of comment as 
```
keccak256("LimitOrder(uint8 orderType,uint256 salt,address trader,address baseToken,bool isBaseToQuote,bool isExactInput,uint256 amount,uint256 oppositeAmountBound,uint256 deadline,uint160 sqrtPriceLimitX96,bytes32 referralCode,bool reduceOnly,uint80 roundIdWhenCreated,uint256 triggerPrice)")
```
and assign the value as `54ea8c184890b5a7f7321f45a2ae952f0af50e3467b2216418a683566bf57c30` directly **AND open a ticket for the deployer to check them being the same before deploying**!

Reply:
You mean to save the gas when deploying or when filling orders? Or both?

Reply:
Both!

Reply:
```
bytes32 public constant LIMIT_ORDER_TYPEHASH = 0x54ea8c184890b5a7f7321f45a2ae952f0af50e3467b2216418a683566bf57c30;
```

1. No double quote
2. Prefix `0x`